### PR TITLE
Master loongson

### PR DESCRIPTION
**Pull request by zhanglixia-hf**
We have completed all the MMI optimizations that we have planned for the Loongson platform.

### CODE REVIEW

**Comment by zhanglixia-hf** [comment]
I think the Travis CI error is a test environment problem, can you retest or check the test environment?

**Comment by dcommander** [comment]
Thank you for the contribution.  I will integrate this into the evolving libjpeg-turbo 2.1.x code base whenever I get a chance.  It may take some time before I am able to do that, as I am out of money right now and am having to focus on paying projects until the end of this year.  Be patient.

**Comment by zhanglixia-hf** [comment]
Ok, I will wait patiently. If you have time, if you find any problems with this patch, please contact me.

File: simd/loongson/loongson-mmintrin.h [hunk 1]
```diff
@@ -1245,6 +1253,36 @@ _mm_load_si64(const __m64 *src)
...
+      : "=f" (ret)
+      : "r" (src)
+      : "memory"
+     );
+
+  return ret;
+}
+
+extern __inline __m64 FUNCTION_ATTRIBS
+_mm_loadgs_si64(const __m64 *src)
```
Comment (FlyGoat) [inline simd/loongson/loongson-mmintrin.h]: Why you declare the same function two times?

**Comment by FlyGoat** [comment]
Hi there
Thanks for your works on Loongson
After fix a obvious build error, the PR faild a lot of test on Loongson-3B1500 with Lemote Fedora28, with mostly SIGTRAP or illegal instructions.
Please have a check 
66% tests passed, 104 tests failed out of 302

Total Test time (real) =  24.43 sec

The following tests FAILED:
	  1 - tjunittest-shared (SIGTRAP)
	  2 - tjunittest-shared-alloc (SIGTRAP)
	  3 - tjunittest-shared-yuv (ILLEGAL)
	  4 - tjunittest-shared-yuv-alloc (ILLEGAL)
	  5 - tjunittest-shared-yuv-nopad (ILLEGAL)
	 10 - tjbench-shared-tile-420-8x8-cmp (Failed)
	 11 - tjbench-shared-tile-422-8x8-cmp (Failed)
	 12 - tjbench-shared-tile-444-8x8-cmp (Failed)
	 14 - tjbench-shared-tile-420-16x16-cmp (Failed)
	 15 - tjbench-shared-tile-422-16x16-cmp (Failed)
	 16 - tjbench-shared-tile-444-16x16-cmp (Failed)
	 18 - tjbench-shared-tile-420-32x32-cmp (Failed)
	 19 - tjbench-shared-tile-422-32x32-cmp (Failed)
	 20 - tjbench-shared-tile-444-32x32-cmp (Failed)
	 22 - tjbench-shared-tile-420-64x64-cmp (Failed)
	 23 - tjbench-shared-tile-422-64x64-cmp (Failed)
	 24 - tjbench-shared-tile-444-64x64-cmp (Failed)
	 26 - tjbench-shared-tile-420-128x128-cmp (Failed)
	 27 - tjbench-shared-tile-422-128x128-cmp (Failed)
	 28 - tjbench-shared-tile-444-128x128-cmp (Failed)
	 31 - tjbench-shared-tile-420m-8x8-cmp (Failed)
	 32 - tjbench-shared-tile-422m-8x8-cmp (Failed)
	 33 - tjbench-shared-tile-420m-16x16-cmp (Failed)
	 34 - tjbench-shared-tile-422m-16x16-cmp (Failed)
	 35 - tjbench-shared-tile-420m-32x32-cmp (Failed)
	 36 - tjbench-shared-tile-422m-32x32-cmp (Failed)
	 37 - tjbench-shared-tile-420m-64x64-cmp (Failed)
	 38 - tjbench-shared-tile-422m-64x64-cmp (Failed)
	 39 - tjbench-shared-tile-420m-128x128-cmp (Failed)
	 40 - tjbench-shared-tile-422m-128x128-cmp (Failed)
	 55 - djpeg-shared-422-ifast-cmp (Failed)
	 57 - djpeg-shared-422m-ifast-cmp (Failed)
	 65 - djpeg-shared-420-q100-ifast-prog-cmp (Failed)
	 67 - djpeg-shared-420m-q100-ifast-prog-cmp (Failed)
	 83 - djpeg-shared-3x2-float-prog-cmp (Failed)
	 87 - djpeg-shared-3x2-ifast-prog-cmp (Failed)
	 95 - djpeg-shared-420m-ifast-ari-cmp (Failed)
	 99 - djpeg-shared-420m-islow-2_1-cmp (Failed)
	101 - djpeg-shared-420m-islow-15_8-cmp (Failed)
	109 - djpeg-shared-420m-islow-7_8-cmp (Failed)
	111 - djpeg-shared-420m-islow-3_4-cmp (Failed)
	113 - djpeg-shared-420m-islow-5_8-cmp (Failed)
	115 - djpeg-shared-420m-islow-1_2-cmp (Failed)
	117 - djpeg-shared-420m-islow-3_8-cmp (Failed)
	121 - djpeg-shared-420m-islow-1_8-cmp (Failed)
	123 - djpeg-shared-420-islow-256-cmp (Failed)
	133 - djpeg-shared-420-islow-skip15_31-cmp (Failed)
	135 - djpeg-shared-420-islow-ari-skip16_139-cmp (Failed)
	138 - djpeg-shared-420-islow-prog-crop62x62_71_71-cmp (Failed)
	143 - djpeg-shared-444-islow-skip1_6-cmp (Failed)
	146 - djpeg-shared-444-islow-prog-crop98x98_13_13-cmp (Failed)
	149 - djpeg-shared-444-islow-ari-crop37x37_0_0-cmp (Failed)
	152 - tjunittest-static (SIGTRAP)
	153 - tjunittest-static-alloc (SIGTRAP)
	154 - tjunittest-static-yuv (ILLEGAL)
	155 - tjunittest-static-yuv-alloc (ILLEGAL)
	156 - tjunittest-static-yuv-nopad (ILLEGAL)
	161 - tjbench-static-tile-420-8x8-cmp (Failed)
	162 - tjbench-static-tile-422-8x8-cmp (Failed)
	163 - tjbench-static-tile-444-8x8-cmp (Failed)
	165 - tjbench-static-tile-420-16x16-cmp (Failed)
	166 - tjbench-static-tile-422-16x16-cmp (Failed)
	167 - tjbench-static-tile-444-16x16-cmp (Failed)
	169 - tjbench-static-tile-420-32x32-cmp (Failed)
	170 - tjbench-static-tile-422-32x32-cmp (Failed)
	171 - tjbench-static-tile-444-32x32-cmp (Failed)
	173 - tjbench-static-tile-420-64x64-cmp (Failed)
	174 - tjbench-static-tile-422-64x64-cmp (Failed)
	175 - tjbench-static-tile-444-64x64-cmp (Failed)
	177 - tjbench-static-tile-420-128x128-cmp (Failed)
	178 - tjbench-static-tile-422-128x128-cmp (Failed)
	179 - tjbench-static-tile-444-128x128-cmp (Failed)
	182 - tjbench-static-tile-420m-8x8-cmp (Failed)
	183 - tjbench-static-tile-422m-8x8-cmp (Failed)
	184 - tjbench-static-tile-420m-16x16-cmp (Failed)
	185 - tjbench-static-tile-422m-16x16-cmp (Failed)
	186 - tjbench-static-tile-420m-32x32-cmp (Failed)
	187 - tjbench-static-tile-422m-32x32-cmp (Failed)
	188 - tjbench-static-tile-420m-64x64-cmp (Failed)
	189 - tjbench-static-tile-422m-64x64-cmp (Failed)
	190 - tjbench-static-tile-420m-128x128-cmp (Failed)
	191 - tjbench-static-tile-422m-128x128-cmp (Failed)
	206 - djpeg-static-422-ifast-cmp (Failed)
	208 - djpeg-static-422m-ifast-cmp (Failed)
	216 - djpeg-static-420-q100-ifast-prog-cmp (Failed)
	218 - djpeg-static-420m-q100-ifast-prog-cmp (Failed)
	234 - djpeg-static-3x2-float-prog-cmp (Failed)
	238 - djpeg-static-3x2-ifast-prog-cmp (Failed)
	246 - djpeg-static-420m-ifast-ari-cmp (Failed)
	250 - djpeg-static-420m-islow-2_1-cmp (Failed)
	252 - djpeg-static-420m-islow-15_8-cmp (Failed)
	260 - djpeg-static-420m-islow-7_8-cmp (Failed)
	262 - djpeg-static-420m-islow-3_4-cmp (Failed)
	264 - djpeg-static-420m-islow-5_8-cmp (Failed)
	266 - djpeg-static-420m-islow-1_2-cmp (Failed)
	268 - djpeg-static-420m-islow-3_8-cmp (Failed)
	272 - djpeg-static-420m-islow-1_8-cmp (Failed)
	274 - djpeg-static-420-islow-256-cmp (Failed)
	284 - djpeg-static-420-islow-skip15_31-cmp (Failed)
	286 - djpeg-static-420-islow-ari-skip16_139-cmp (Failed)
	289 - djpeg-static-420-islow-prog-crop62x62_71_71-cmp (Failed)
	294 - djpeg-static-444-islow-skip1_6-cmp (Failed)
	297 - djpeg-static-444-islow-prog-crop98x98_13_13-cmp (Failed)
	300 - djpeg-static-444-islow-ari-crop37x37_0_0-cmp (Failed)

**Comment by zhanglixia-hf** [comment]
I resubmit the patch to https://github.com/libjpeg-turbo/libjpeg-turbo/pull/306。

**Pull request closed without merging**